### PR TITLE
Clear lints when a file is deleted

### LINKED ISSUE #92: VSCode extension keeps errors and warnings from deleted files
If a file with errors in it is deleted, those errors will remain in the Problems menu until the window is reloaded.

### PR DESCRIPTION
**Pull request by LouieK22**
Closes #92

### CODE REVIEW

File: selene-vscode/src/extension.ts [hunk 1]
```diff
@@ -121,9 +121,16 @@ export async function activate(context: vscode.ExtensionContext) {
...
+    async function clearLints(documentUri: vscode.Uri) {
+        diagnosticsCollection.set(documentUri, [])
+    }
+
     vscode.workspace.onDidSaveTextDocument(lint)
     vscode.workspace.onDidOpenTextDocument(lint)
     vscode.workspace.onDidChangeTextDocument(event => lint(event.document))
+    vscode.workspace.onWillDeleteFiles(event => {
+        event.files.forEach(clearLints);
+    })
```
Comment (Kampfkarren) [inline selene-vscode/src/extension.ts:133]: I think having the `clearLints` function separate from the only place it's being called hurts readability. This should just be changed to a `for (const ... of ...)` loop.

**Comment by LouieK22** [comment]
Dependency updates are needed since the API for onWillDeleteFiles didn't exist on the current version.

**Comment by Kampfkarren** [comment]
Makes sense then, will test this on my own machine tomorrow.

**Comment by Kampfkarren** [comment]
image
Doesn't seem to work for me?

**Comment by Kampfkarren** [comment]
Can you add a CHANGELOG entry under an `Unreleased` header?

**Comment by Kampfkarren** [comment]
Thanks!

**Pull request merged**